include/SSVOpenHexagon/Core/HexagonGame.hpp

```diff
@@ -262,12 +262,14 @@ class HexagonGame
 
     [[nodiscard]] float getSpeedMultDM() const noexcept
     {
-        return levelStatus.speedMult * (std::pow(difficultyMult, 0.65f));
+        const auto res = levelStatus.speedMult * (std::pow(difficultyMult, 0.65f));
+        return (res < levelStatus.speedMax && levelStatus.speedMax > 0.f) ? res : levelStatus.speedMax;
```

Comment:
If `levelStatus.speedMax` is zero, this line seems to always return `levelStatus.speedMax`, which seems wrong. Is that intended? Or should it be something like:

```cpp
if (levelStatus.speedMax <= 0.f)
{
   return res; 
}

return std::clamp(res, 0.f, levelStatus.speedMax);
```

?